aospy/utils/io.py

```diff
@@ -38,17 +35,6 @@ def get_parent_attr(obj, attr, strict=False):
             return None
 
 
-def dict_name_keys(objs):
-    """Create dict whose keys are the 'name' attr of the objects."""
-    assert isinstance(objs, (tuple, list, dict, set))
-    if isinstance(objs, (tuple, list, set)):
-        try:
-            return {obj.name: obj for obj in objs}
-        except AttributeError as e:
-            raise AttributeError(e)
-    return objs
-
-
 def to_dup_list(x, n, single_to_list=True):
```

Comment:
FYI I recently learned of the [vulture](https://github.com/myint/vulture) package that finds unused code in a package, could be useful to use this periodically on aospy